src/main/java/edu/umich/tl/CourseDelete.java

```diff
@@ -509,6 +540,63 @@ private static void deleteTheCourse(Course course, ApiCallHandler apiHandler,Cou
 		coursesForDelete.addDeletedCourse(course);
 	}
 	
+	private static void sendAnEmailReport(ArrayList<Term> previousTerms, CoursesForDelete coursesForDelete) {
+		M_log.debug("sendAnEmailReport(): Called");
+
+		Properties properties = System.getProperties();
+		properties.put(MAIL_SMTP_AUTH, "false");
+		properties.put(MAIL_SMTP_STARTTLS, "true"); 
+		properties.put(MAIL_SMTP_HOST, mailHost);
+		properties.put(MAIL_DEBUG_PROPERTY, mailDebug); //true=debugEnabled
+
+		Session session = Session.getInstance(properties);
+		MimeMessage message = new MimeMessage(session);
+
+		try {
+			ArrayList<String> termsList = new ArrayList<String>();
+			for (Term term : previousTerms) {
+				termsList.add(term.getTermName());
+			}
+			message.addRecipient(Message.RecipientType.TO, new InternetAddress(toEmailAddress));
+			message.setSubject(String.format(emailSubjectText,termsList));
+
+			StringBuilder msgBody=new StringBuilder();
+			msgBody.append("Total unpublished courses for terms ");
+			msgBody.append(termsList.toString());
+			msgBody.append(" =>");
+			msgBody.append(coursesForDelete.getCourses().size());
+			msgBody.append("\n");
+			msgBody.append("Courses Deleted => ");
+			msgBody.append(coursesForDelete.getDeletedUnpublishedCourses().size());
+			BodyPart msgBodyPart = new MimeBodyPart();
+			msgBodyPart.setText(msgBody.toString());
+
+			Multipart multipart = new MimeMultipart();
+			multipart.addBodyPart(msgBodyPart);
+			msgBodyPart=new MimeBodyPart();
+			msgBodyPart.setDataHandler(new DataHandler(new ByteArrayDataSource(generateCSVFileData(coursesForDelete).getBytes(), TEXT_CSV)));
+			msgBodyPart.setFileName(String.format(reportFileName, termsList));
+			multipart.addBodyPart(msgBodyPart);
+			message.setContent(multipart);
+			M_log.info("Sending mail...");
+			Transport.send(message);
+		} catch(MessagingException e) {
+			M_log.error("Email notification failed ",e);
```

Comment:
If it fails may want to log the data on a failure if it isn't too big. Something should be done / logged if losing the information would cause a serious issue.  

Is there a limit on attachment size?  Maybe just within UoM?  If the file is likely to be very large maybe the results should be broken up in to chunks.